site/topics/functions/functions.rst

```diff
@@ -237,42 +237,54 @@ Abstraction
     * For example, when driving a car, do you think about the pistons firing?
 
 
-Back to concrete things...
-==========================
+Format of a Function
+====================
 
-* The general format for defining a function is::
+* The general format for defining a function is as follows
+
+.. code-block:: python
+    :linenos:
+
+    def function_name(parameter_1, parameter_2, parameter_3, parameter_4, ... parameter_n):
+        statement 1
+        statement 2
+        ...
+        statement m
+
+
+* ``function_name`` is what the function is named
+* ``parameter_1, parameter_2`` , etc. are called the parameters, you can have as many as you like and call them almost whatever you want
+
+    * You can think of these like variables
+
+* You tell Python which statements make up the body of the function by using *indentation*
+
+    * Some languages use other syntax, like ``begin`` and ``end`` or braces (``{ }``)
 
-	def function_name(p1, p2, p3, p4, ... ):
-		statement 1
-		statement 2
-		...
-		statement m
-		
-* ``function_name`` is... the name of the function. This can be almost whatever you want.
-* ``p1, p2`` , etc. are called the *parameters*, you can have as many as you like and call them almost whatever you want. 
-* You tell Python which statements make up the *body* of the function by using *indentation*.
-    * This is a somewhat unique feature of Python. 
-        * And somewhat hated by some people. 
-    * Many other languages use pairs like ``begin, end`` , ``do, done`` or ``{, }`` to delimit the body of a function.
 
 .. admonition:: Activity
 
-    Write a function ``catstr`` which takes two strings as parameters and then prints out the concatenation of the strings. e.g., if I call ``catstr('Hello ','world!')`` it will print ``Hello world!``.
+    Write a function ``concatenate_strings`` that takes two strings as parameters and then returns the concatenation of
+    the strings. For example, if I call ``concatenate_strings("Hello","World!")`` it will return ``Helloworld!``. Do
```

Comment:
Should be `HelloWorld!` -- capital `W`

Reply:
1. I originally had it that way
2. Changed it to emphasize that there is no space between them
3. I later added the thing saying "don't add a space"

In other words, I'll put it back --- good catch.